proposals/003-metric-improvenents.md

```diff
@@ -0,0 +1,262 @@
+metrics use-cases where the requests sent by the filters themselves need to be understood.
+
+### New distribution `kroxylicious_downstream_kafka_message_size_bytes`
+
+`kroxylicious_downstream_kafka_message_size_bytes` records the message size of every message passing to/from the downstream.
+It is discriminated by `virtual_cluster`, `node_id`, `flowing`, `opaque`, `api_key`, and `api_version` labels.  `node_id` will only
+be present if for requests going to or responses coming from a broker.  For bootstrap interactions, `node_id` will be
+absent.
+
+Use cases that were previously served by the deprecated metric `kroxylicious_payload_size_bytes` can now use this metric
+instead. The new metric records the size of both **opaque** and **decoded** messages allowing the user to understand the
+number of bytes traversing the proxy.
+
+### New distribution `kroxylicious_upstream_kafka_message_size_bytes`
+
+`kroxylicious_upstream_kafka_message_size_bytes` records the message size of every message passing to/from the upstream.
+It is discriminated by `virtual_cluster`, `node_id`, `flowing`, `opaque`, `api_key`, `api_version` and `originating_filter` labels.
+
+`node_id` will only be present if for requests going to or responses coming from a broker.  For bootstrap interactions, `node_id` will be
+absent.
+
+`originating_filter` is only used for requests sent by the filters themselves (i.e. the `sendRequest` API)
+and their responses.  It will be populated with the filter's name.  Otherwise `originating_filter` will be absent.
+
+### New distribution `kroxylicious_message_process_time`
+
+This metric records the length of time a message (request or response) has taken to traverse the proxy.
+
+* The start time will be time the bytes containing the message arrived from the network (in the Netty handler, before any decoding).
+* The end time will be the time that the network write at the proxy's opposite end completes (i.e. the Netty write promise completes)
+
+The use cases supported by this metric are ones where you are interested in how much processing time it being incurred 
+by the proxy decoding and encoding messages and any processing time incurred by the filter.
+
+The requests sent by filter themselves, the start time for a request will be time filter called `#sendRequest`. The end time
+will be the time the network write at the proxy upstream completes.  For responses, the start time is time the bytes containing
+the message arrived from the network. The end time will be the time the response reaches the originating filter (i.e. the future completes).
+
+### New gauges `kroxylicious_(up|down)stream_backpressure_state`
+
+Reports the backpressure state for upstream and downstream.  The use-case for metric is to allow the user to
+understand when the proxy must pause reading owing whilst asynchronous work completes (for example, in record validation
```

Comment:
I'm not sure what `owing` is doing in the sentence. I presume its a hang over from a previous draft.